project/src/main/webapp/gmail.js

```diff
@@ -114,36 +154,20 @@ function getMessage(messageId) {
     messageRequest.execute(response => {
         //Convert from base64 encoding to text.
         const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
-        var hasFormatFlowed = false;
-        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
-        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
-        var emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
-        if (emailBodyValue.indexOf("format=flowed") != -1) {
-            hasFormatFlowed = true;
-        }
-        emailBodyValue = emailBodyValue.substring(200);
-        const emailDateStartIndex = decodedEmail.indexOf("Date:");
-        const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
-        var emailDateValue = decodedEmail.substring(emailDateStartIndex, emailDateStartIndex + emailDateEndIndex);
-        const MessageIDStartIndex = decodedEmail.indexOf("Message-ID:") + 12;
-        const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
-        var messageID = decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex);
-
-        //Grabs the date of the email.
-        if (emailDateValue.indexOf("X-Notifications") != -1) {
-            emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
-        }
-
-        //A condition to check if it's an actual email.
-        if (!hasFormatFlowed && emailDateValue.length > 1 && emailBodyValue.indexOf("Daily Insider") == -1 /*(!(messageID in emailObject))*/) {
-            emailObject[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+        const emailBodyValue = getEmailBody(decodedEmail);
```

Comment:
What happens when there is no response from the Gmail API?

Reply:
If there is no response from the Gmail API, then the messageID is undefined.